kythe/cxx/indexer/proto/testdata/proto_verifier_test.bzl

```diff
@@ -14,31 +14,53 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@bazel_skylib//lib:collections.bzl", "collections")
 load(
     "@io_kythe//tools/build_rules/verifier_test:verifier_test.bzl",
     "KytheVerifierSources",
     "extract",
     "index_compilation",
     "verifier_test",
 )
+load("@rules_proto//proto:defs.bzl", "ProtoInfo")
 
 def _invoke(rulefn, name, **kwargs):
     """Invoke rulefn with name and kwargs, returning the label of the rule."""
     rulefn(name = name, **kwargs)
     return "//{}:{}".format(native.package_name(), name)
 
+def get_proto_files_and_proto_path_opts(protolibs):
+    """Given a list of proto_library targets, returns a list of all transitively
+    -included .proto files and a list of options to pass to protoc (containing
+    --proto_path locations).
+    """
+    srcs = []
+    proto_paths = []
+    for lib in protolibs:
+        if ProtoInfo not in lib:
+            fail("Expected a proto_library, got: %s" % str(lib))
+        for src in lib[ProtoInfo].transitive_sources.to_list():
+            if src.path.endswith(".proto"):
+                srcs.append(src)
+            proto_paths += lib[ProtoInfo].transitive_proto_path.to_list()
```

Comment:
You probably want to collect these into a depset(), rather than listifying them.

```
for ... :
  proto_paths.append(transitive_proto_path)
...
proto_paths = depset(transitive = proto_paths)
```

And then you can use Args rather than string options: https://docs.bazel.build/versions/master/skylark/lib/attr.html#label

Reply:
Also, why are you adding the same proto_paths for each src in `lib[ProtoInfo].transitive_sources` ?

Reply:
You *probably* only want to use the direct sources for the verifier facts.  It's not a huge deal, but verifier runtime is superlinear in the number of facts, so avoiding potentially redundantly checking them transitively might be good.

Reply:
Good point about only using direct sources for the verifier facts. I changed the helper function to now return a list of toplevel sources alongside the full set of sources. The toplevel sources are passed as sources to the extractor, while the full set of imported protos is given as `deps`.

I tried using a depset for `proto_paths`, but it later gets passed to the `extract()` rule from `verifier_test.bzl`, which accepts `opts` as a list. I could use a depset and convert it back, but not sure if it's worth the extra shuffling.

Reply:
It also accepts `Args` (just not a list-of-args, which is probably a mistake).

Reply:
Thanks, I got it working by passing an `Args` directly (not in a list).